### PR TITLE
Rename metrics/logs builders to `hints`

### PR DESCRIPTION
**Pull request by exekias**
This change renames current `logs` and `metrics` builders to `hints`. These builders are beat specific and are minded to create configs based on hints (from labels or annotations). Config would then look like this:

```
 metricbeat.autodiscover:
   providers:
     - type: kubernetes
       builders:
        - type: hints
```

It also introduces a shortcut to configure just the `hints` builder, which should cover most use cases for common users:

```
metricbeat.autodiscover:
  providers:
    - type: kubernetes
      hints.enabled: true
```

No changelog is needed as this is updating hints based autodiscover, which is unreleased

### CODE REVIEW

**Comment by exekias** [comment]
Yes, if you want advanced use cases like that one, you can just remove `hints.enabled` and define the set of builders. I've added an "Advanced usage" topic inside autodiscover docs for that

File: filebeat/docs/autodiscover-hints.asciidoc [hunk 1]
```diff
@@ -38,8 +37,18 @@ include_lines
 
 [float]
 ==== Docker
-Docker allows providing hints through Docker labels. When containers are spun up the Builder understands
-labels that convey hints to convey additional configuration to the input.
+
+Docker autodiscover provider supports hints too, it's enabled in the same way:
```
Comment (ruflin) [inline filebeat/docs/autodiscover-hints.asciidoc]: I would not use `too` here as people might jump to this section of the docs directly.

File: libbeat/autodiscover/providers/docker/docker.go [hunk 2]
```diff
@@ -48,22 +48,14 @@ func AutodiscoverBuilder(bus bus.Bus, c *common.Config) (autodiscover.Provider,
...
 	}
 
-	var builders autodiscover.Builders
-	for _, bcfg := range config.Builders {
-		builder, err := autodiscover.Registry.BuildBuilder(bcfg)
-		if err != nil {
-			return nil, err
-		}
-		builders = append(builders, builder)
+	builders, err := autodiscover.NewBuilders(config.Builders, config.HintsEnabled)
```
Comment (ruflin) [inline libbeat/autodiscover/providers/docker/docker.go]: Nice simplification of the code.

File: libbeat/docs/shared-autodiscover.asciidoc [hunk 3]
```diff
@@ -119,35 +114,52 @@ contain variables from the autodiscover event. They can be accessed under data n
...
 
+[[configuration-autodiscover-hints]]
+=== Hints based autodiscover
+
+include::../../{beatname_lc}/docs/autodiscover-hints.asciidoc[]
+
+[[configuration-autodiscover-advanced]]
+=== Advanced usage
+
+////
```
Comment (ruflin) [inline libbeat/docs/shared-autodiscover.asciidoc]: Leftover comment?

File: metricbeat/tests/system/test_autodiscover.py [hunk 4]
```diff
@@ -86,9 +86,4 @@ def test_docker_labels(self):
         container.stop()
         self.wait_until(lambda: self.log_contains('Autodiscover stopping runner: memcached'))
 
-        output = self.read_output_json()
```
Comment (ruflin) [inline metricbeat/tests/system/test_autodiscover.py]: Is this check not needed anymore?

File: libbeat/docs/shared-autodiscover.asciidoc (hunk 3 above)
Comment (exekias) [reply libbeat/docs/shared-autodiscover.asciidoc]: I left this commented as I expect new builders coming soon, they will be documented there (including hints) once we have more than just 1.

File: metricbeat/tests/system/test_autodiscover.py (hunk 4 above)
Comment (exekias) [reply metricbeat/tests/system/test_autodiscover.py]: I removed it as redundant, but on a second thought, it makes sense to double check here, I'll add it again

Comment (exekias) [reply metricbeat/tests/system/test_autodiscover.py]: I've been fighting them a little more, hopefully they work now, without sleeps

Comment (exekias) [reply metricbeat/tests/system/test_autodiscover.py]: It worked :+1:

**Comment by exekias** [comment]
jenkins, retest this please

**Pull request merged**